OpenDreamClient/Rendering/DreamIcon.cs

```diff
@@ -62,6 +62,7 @@ internal sealed class DreamIcon(IGameTiming gameTiming, IClyde clyde, ClientAppe
     public void Dispose() {
         _ping?.Dispose();
         _pong?.Dispose();
+        DMI?.onUpdateCallbacks.Remove(DirtyTexture);
```

Comment:
Can you instead do `DMI = null` here and move this `Remove()` to `DMI`'s setter? That way the callback gets removed on any change to `DMI`.